src/main/webapp/app/utils/date.utils.ts

```diff
@@ -0,0 +1,18 @@
+import dayjs from 'dayjs/esm';
+
+/**
+ * Converts a date in order to send it to the server
+ * @param date the date either as proper date or string representation
+ */
+export function convertDateFromClient(date?: dayjs.Dayjs): string | undefined {
+    // Result completionDate is a dayjs object -> toJSON.
+    return date && dayjs.isDayjs(date) ? date.toJSON() : date;
```

Comment:
Isn't it necessary to check if date.isValid() like it was done previously (in some cases) ?

Reply:
You're right. This is now also included in the utility function